synfig-studio/src/synfigapp/actions/waypointsetsmart.cpp

```diff
@@ -128,6 +135,14 @@ Action::WaypointSetSmart::set_param(const synfig::String& name, const Action::Pa
 
 		return static_cast<bool>(value_node);
 	}
+
+	if(name=="value_desc" && param.get_type()==Param::TYPE_VALUEDESC)
+	{
+		value_desc=param.get_value_desc();
+
+		return true;
+	}
```

Comment:
You should check if param.get_value_desc() is a valid ValueDesc (it should refer to a value node, right?)
So, only store it to `value_desc` if it is a valid one.

Reply:
hmm I tried to do so by checking `parent_is_value_node()` but it always return false. Any suggestions ?

Reply:
`is_valuenode()` is enough (I just checked `get_valuenode()` implementation ;) )

Reply:
hmm `is_value_node` is giving issues (probably due to the extra stuff it calls other than `is_valid`). while just the `!` operator was working (it only checks `is_valid`). What do you think ?

Reply:
`is_valid()` isn't enough, because you need (line 124):
``` c++
!(LinkableValueNode::Handle::cast_dynamic(x.find("value_desc")->second.get_value_desc().get_value_node())));
```

Reply:
Then how about something like 
`if (!value_desc || !value_desc.get_value_node())` then we return false.
because using `is_value_node` was just constantly returning false.

Reply:
> `is_valid()` isn't enough, because you need (line 124):
> 
> ```c++
> !(LinkableValueNode::Handle::cast_dynamic(x.find("value_desc")->second.get_value_desc().get_value_node())));
> ```

Hey @rodolforg, I got a little carried away by some stuff but I looked back into this. 

On line 124 I am basically checking to see if this `valuedesc` is referring to a linkable valuenode. If it is then it referring to a an expandable parameter to which add waypoint shouldn't be applied to. What I mean is when `get_value_node()` returns  a null pointer the cast fails which is what is required anyways so `get_value_node()` doesn't necessarily have to return something valid.

Now going back to our topic.For the very first waypoint added the `value_desc` doesnt contain a valuenode, so using `get_value_node()` or `is_value_node()` always returns false the very first time a waypoint is added to a parameter. 

This is why I think it's sufficient to just check with is_valid() because we really don't need the value desc to refer to a valuenode. If its a parameter without any waypoints it initially doesnt but then when it has waypoints it does.

Reply:
@rodolforg ping ^_^.

Reply:
So, shouldn't you check ifnit isn't a valuenode?

Reply:
> So, shouldn't you check ifnit isn't a valuenode?

oh ok adding a check to make sure it isn't a valuenode would work.

Reply:
Done.
At first I was a bit confused as why it might be needed. But I think it makes sense now, if it is not the first time (i.e. value desc is a value node) the we don't need the value desc because we will not need a value desc set action.